plugins/GranularPitchShifter/GranularPitchShifter.cpp

```diff
@@ -0,0 +1,269 @@
+/*
+ * GranularPitchShifter.cpp
+ *
+ * Copyright (c) 2024 Lost Robot <r94231/at/gmail/dot/com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "GranularPitchShifter.h"
+
+#include <cmath>
+#include "embed.h"
+#include "plugin_export.h"
+
+
+namespace lmms
+{
+
+extern "C"
+{
+Plugin::Descriptor PLUGIN_EXPORT granularpitchshifter_plugin_descriptor =
+{
+	LMMS_STRINGIFY(PLUGIN_NAME),
+	"Granular Pitch Shifter",
+	QT_TRANSLATE_NOOP("PluginBrowser", "Granular pitch shifter"),
+	"Lost Robot <r94231/at/gmail/dot/com>",
+	0x0100,
+	Plugin::Type::Effect,
+	new PluginPixmapLoader("logo"),
+	nullptr,
+	nullptr,
+} ;
+}
+
+
+GranularPitchShifterEffect::GranularPitchShifterEffect(Model* parent, const Descriptor::SubPluginFeatures::Key* key) :
+	Effect(&granularpitchshifter_plugin_descriptor, parent, key),
+	m_granularpitchshifterControls(this),
+	m_prefilter({PrefilterLowpass(), PrefilterLowpass()})
+{
+	autoQuitModel()->setValue(autoQuitModel()->maxValue());
+	
+	changeSampleRate();
+}
+
+
+bool GranularPitchShifterEffect::processAudioBuffer(sampleFrame* buf, const fpp_t frames)
+{
+	if (!isEnabled() || !isRunning()) { return false; }
+
+	const float d = dryLevel();
+	const float w = wetLevel();
+	
+	const ValueBuffer * pitchBuf = m_granularpitchshifterControls.m_pitchModel.valueBuffer();
```

Comment:
Just so you know, valuebuffer is on it's way out in #7170. You don't need to worry for now. I'll intervene for fixing this PR up if that one gets merged.